ls1intum/Artemis
# Move exercise hint to new course management routes

### Checklist

### Motivation and Context


The exercise are still managed under the old routes. They should be moved to the new course management routes.

### Description

- Split up hint management and participation components
- Import hint management components in course management module
- Update routes of management components

Before the changes, the hint module got loaded with the code-editor, meaning that you always had to open the editor at least once if you wanted to change or add any hints to a course. We should also have this problem on production. By splitting up the hint modules and having a new dependency in the course management module, we fix this problem.

### Steps for Testing


1. Log in to Artemis
2. Navigate to a programming exercise on the course administration
3. `View` the exercise and edit the hints
4. You should be able to add, edit, delete and view hints just like before (and without having to open the code-editor once)
